scripts/list_boards.py

```diff
@@ -6,8 +6,132 @@
 import argparse
 from collections import defaultdict
 import itertools
-from pathlib import Path
+from pathlib import Path, PurePath
+import pykwalify.core
 from typing import NamedTuple
+import sys
+import yaml
+
+METADATA_SCHEMA = '''
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The boards.yml file is a simple list of key value pairs containing boards
+# and their location which is used by the build system.
+type: map
+mapping:
+  boards:
+    required: true
+    type: seq
+    sequence:
+      - type: map
+        mapping:
+          name:
+            required: true
+            type: str
+            desc: Name of the board
+          folder:
+            required: true
+            type: str
+            desc: Location of the board implementation relative to the boards.yml file.
+          vendor:
+            required: false
+            type: str
+            desc: SoC family of the SoC on the board.
+                  This field is of informational use and can be used for filtering of boards.
+          family:
+            required: false
+            type: str
+            desc: SoC family of the SoC on the board.
+                  This field is of informational use and can be used for filtering of boards.
+          arch:
+            required: false
+            type: str
+            desc: architecture of the SoC on a single SoC board, for a board with multiple SoCs
+                  or multi-core SoC with mixed architectures, the value mixed can be used.
+                  This field is of informational use and can be used for filtering of boards.
+          comment:
+            required: false
+            type: str
+            desc: Free form comment with extra information regarding the board.
+'''
+
+BOARD_METADATA_SCHEMA = '''
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The board.yml file is a simple list of key value pairs containing board
+# information like: name, vendor, arch, cpusets.
+type: map
+mapping:
+  board:
+    required: true
+    type: map
+    mapping:
+      name:
+        required: true
+        type: str
+        desc: Name of the board
+      vendor:
+        required: false
+        type: str
+        desc: SoC family of the SoC on the board.
+      family:
+        required: false
+        type: str
+        desc: SoC family of the SoC on the board.
+      arch:
+        required: false
+        type: str
+        desc: architecture of the SoC on a single SoC board, for a board with multiple SoCs
+              or multi-core SoC with mixed architectures, the value mixed can be used.
```

Comment:
(for vender and arch) I think these should be mandatory, just because I do not think we should accept **any** soc/arch/board PRs to zephyr that lack this fundamental information

Reply:
i'm fine with having this as a project requirement for consistency, although from technical perspective it's not required.

Let's get opinion at the arch wg meeting.